src/@next/components/organisms/StripePaymentGateway/StripePaymentGateway.tsx

```diff
@@ -66,29 +104,108 @@ const StripePaymentGateway: React.FC<IProps> = ({
             firstDigits: null,
             lastDigits: card?.last4,
           });
+          setPaymentMethod(payload.paymentMethod);
         }
       } else {
         const stripePayloadErrors = [
-          {
-            message:
-              "Payment submission error. Stripe gateway returned no payment method in payload.",
-          },
+          new Error(
+            intl.formatMessage(stripeErrorMessages.paymentSubmissionError)
+          ),
         ];
         setSubmitErrors(stripePayloadErrors);
         onError(stripePayloadErrors);
       }
     } else {
       const stripeElementsErrors = [
-        {
-          message:
-            "Stripe gateway improperly rendered. Stripe elements were not provided.",
-        },
+        new Error(intl.formatMessage(stripeErrorMessages.geytwayDisplayError)),
       ];
       setSubmitErrors(stripeElementsErrors);
       onError(stripeElementsErrors);
     }
   };
 
+  const handleFormCompleteSubmit = async () => {
+    const stripe = await stripePromise;
+
+    const payment = await submitPayment();
+
+    if (payment.errors?.length) {
+      onError(payment.errors);
+    } else if (!payment?.confirmationNeeded) {
+      submitPaymentSuccess(payment?.order);
+    } else if (!stripe?.confirmCardPayment) {
+      onError([
+        new Error(
+          intl.formatMessage(
+            paymentErrorMessages.cannotHandlePaymentConfirmation
+          )
+        ),
+      ]);
+    } else if (!payment?.confirmationData) {
+      onError([
+        new Error(
+          intl.formatMessage(paymentErrorMessages.paymentNoConfirmationData)
+        ),
+      ]);
+    } else {
+      let paymentAction;
+      try {
+        paymentAction = JSON.parse(
+          payment.confirmationData
+        ) as StripeConfirmationData;
+      } catch (parseError) {
+        onError([
+          new Error(
+            intl.formatMessage(
+              paymentErrorMessages.paymentMalformedConfirmationData
+            )
+          ),
+        ]);
+        return;
+      }
+      if (!paymentMethod?.id) {
+        onError([
+          new Error(
+            intl.formatMessage(stripeErrorMessages.paymentMethodNotCreated)
+          ),
+        ]);
+        return;
+      }
+      let confirmation;
```

Comment:
This could probably be done via const inside the try/catch block, followed by optional chaining wherever a potentially undefined key of this variable is accessed.

Reply:
Not sure if I understand correctly, const or let are scoped variables, so cannot be accessed outside of try/catch. Nevertheless, I have extracted these try/catch to the separate functions with immediate value return and I have added early return instead of if/else in this function, so I hope it is a little bit cleaner now.